### PR TITLE
Collect LimitRange and StorageClass resources

### PR DESCRIPTION
**Pull request by xlucas**
### What does this PR do?

Add collection for `LimitRange` and `StorageClass` resources.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -851,3 +851,5 @@ replace (
 	github.com/prometheus/client_golang => github.com/prometheus/client_golang v1.18.0
 	github.com/prometheus/common => github.com/prometheus/common v0.46.0
 )
+
+replace github.com/DataDog/agent-payload/v5 => github.com/DataDog/agent-payload/v5 v5.0.112-0.20240403100753-82d9646d460f
```
Comment (xlucas) [inline go.mod:855]: note: requires https://github.com/DataDog/agent-payload/pull/297 to be merged before removing this

File: pkg/collector/corechecks/cluster/orchestrator/processors/k8s/limitrange.go [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		GroupId:     pctx.MsgGroupID,
+		GroupSize:   int32(groupSize),
+		LimitRanges: models,
+		Tags:        append(pctx.Cfg.ExtraTags, pctx.ApiGroupVersionTag),
+	}
+}
+
+// ExtractResource is a handler called to extract the resource model out of a raw resource.
+//
+//nolint:revive // TODO(CAPP) Fix revive linter
```
Comment (vboulineau) [inline pkg/collector/corechecks/cluster/orchestrator/processors/k8s/limitrange.go:57]: New code should not have `nolint TODO`

**Review by kangyili — APPROVED** [review]

File: pkg/collector/corechecks/cluster/orchestrator/processors/k8s/limitrange.go (hunk 2 above)
Comment (xlucas) [reply pkg/collector/corechecks/cluster/orchestrator/processors/k8s/limitrange.go:57]: makes sense thanks 👍

**Review by maycmlee — APPROVED** [review]

**Review by vboulineau — APPROVED** [review]

**Review by mx-psi — APPROVED** [review]

**Review by badovm — APPROVED** [review]

**Review by jszwedko — APPROVED** [review]
Approving for files owned by @DataDog/agent-processing-and-routing

**Pull request merged**